src/ACEtk/interpretation/ContinuousEnergyNeutron/Builder/ProbabilityTable/Builder/test/Builder.test.cpp

```diff
@@ -0,0 +1,156 @@
+            aceified |= ranges::action::push_back( captureXS );
+            aceified |= ranges::action::push_back( heating );
+            aceified |= ranges::action::push_back( CDFs );
+            aceified |= ranges::action::push_back( totalXS );
+            aceified |= ranges::action::push_back( elasticXS );
+            aceified |= ranges::action::push_back( fissionXS );
+            aceified |= ranges::action::push_back( captureXS );
+            aceified |= ranges::action::push_back( heating );
+
+            Table::Data data{};
+            pTable.ACEify( data );
+
+            CHECK( ranges::equal( aceified, data.XSS() ) );
+          }
+        }
+      }
+    }
+    WHEN( "adding probability table without heating or fission values" ){
+        tb.interpolationParameter( INT )
+          .inelasticCompetition( inelasticCompetition )
+          .otherAbsorption( otherAbsorption )
+          .factors( factors )
+          .incidentEnergies( njoy::utility::copy( energies ) )
+          .table()
+            .CDFs( njoy::utility::copy( CDFs ) )
+            .totalCrossSections( njoy::utility::copy( totalXS ) )
+            .elasticCrossSections( njoy::utility::copy( elasticXS ) )
+            .captureCrossSections( njoy::utility::copy( captureXS ) )
+          .add()
+          .table()
+            .CDFs( njoy::utility::copy( CDFs ) )
+            .totalCrossSections( njoy::utility::copy( totalXS ) )
+            .elasticCrossSections( njoy::utility::copy( elasticXS ) )
+            .captureCrossSections( njoy::utility::copy( captureXS ) )
+          .add()
+          .table()
+            .CDFs( njoy::utility::copy( CDFs ) )
+            .totalCrossSections( njoy::utility::copy( totalXS ) )
+            .elasticCrossSections( njoy::utility::copy( elasticXS ) )
+            .captureCrossSections( njoy::utility::copy( captureXS ) )
+          .add();
+
+        auto pTable = tb.construct();
+
+        THEN( "the values can be verified" ){
+          CHECK( INT == pTable.interpolationParameter );
+          CHECK( inelasticCompetition == pTable.inelasticCompetition );
+          CHECK( otherAbsorption == pTable.otherAbsorption );
+          CHECK( factors == pTable.factors );
+          CHECK( ranges::equal( energies, pTable.incidentEnergies ) );
+        }
+    }
+  } // GIVEN
+  GIVEN( "invalid inputs" ){
+    WHEN( "incident energies are not all positive" ){
+      std::vector< double > energies{ 1.0, -2.0, 3.0 };
+      THEN( "an exception is thrown" ){
+        CHECK_THROWS_AS( 
+            tb.incidentEnergies( energies ),
+            details::verify::exceptions::NotPositive& 
+        );
+      }
+    }
+    WHEN( "incident energies are not sorted" ){
+      std::vector< double > energies{ 1.0, 4.0, 3.0 };
+      THEN( "an exception is thrown" ){
+        CHECK_THROWS_AS( 
+            tb.incidentEnergies( energies ),
+            details::verify::exceptions::Unsorted& 
+        );
+      }
+    }
```

Comment:
What about problems with the interpolation flag?

Reply:
done